lib/src/Pos/Launcher/Resource.hs

```diff
@@ -289,6 +302,14 @@ allocateNodeContext ancd txpSettings ekgStore = do
     logDebug "Created peersVar"
     mm <- initializeMisbehaviorMetrics ekgStore
 
+    logDebug ("Dequeue policy to core:  " <> (show ((ncDequeuePolicy networkConfig) NodeCore)))
+        `catch` \(e :: SomeException) -> logDebug $ show e
+    logDebug ("Dequeue policy to relay: " <> (show ((ncDequeuePolicy networkConfig) NodeRelay)))
+        `catch` \(e :: SomeException) -> logDebug $ show e
+    logDebug ("Dequeue policy to edge: " <> (show ((ncDequeuePolicy networkConfig) NodeEdge)))
+        `catch` \(e :: SomeException) -> logDebug $ show e
```

Comment:
Please review the 6 lines above also.

Reply:
Catching `SomeException` without immediately throwing it is generally not a good idea, since it can easily swallow arbitrary exceptions and mess up your error handling. You should either catch some specific exception (or class of exceptions) and handle them properly, or -- if, for example, you just want to log that there was some exception -- immediately rethrow the exception.

Also, I'm not sure why you're catching here in the first place. Are you expecting some exceptions from those loggings?

Reply:
https://github.com/input-output-hk/cardano-sl/blob/5db86b391231f3eb655bb90fc879a3cfe7f53f2f/infra/Pos/Infra/Network/Policy.hs#L167-L173

Default dequeue policy of core throws an error for edge nodes.

Reply:
Ok, that makes sense. And since there is nothing in the lines you're using `catch` on that could throw another exception, the usage is ok here (well, technically, `logDebug` itself could throw an exception, but you're also using that on in the code catching the exception.

Reply:
Suggestion: Change the `error` call to a `throw` with an exception type (that isn't string). Then we can catch the specific error.

Alternatively, you can call:

```haskell
foo `catch` \(ErrorCall msg) -> logDebug (pack msg)
```

which will only catch the `error` call.